src/plugins/discover/public/application/apps/main/components/doc_table/components/table_row.tsx

```diff
@@ -58,6 +65,15 @@ export const TableRow = ({
 
   const flattenedRow = useMemo(() => indexPattern.flattenHit(row), [indexPattern, row]);
   const mapping = useMemo(() => indexPattern.fields.getByName, [indexPattern]);
+  const fieldsToShow = useMemo(
+    () =>
+      getFieldsToShow(
+        indexPattern.fields.map((field: IndexPatternField) => field.name),
+        indexPattern,
+        showMultiFields
+      ),
+    [indexPattern, showMultiFields]
+  );
```

Comment:
calculating `getFieldsToShow` here, means, calculating in every row, best way to check this is a adding a `console.count('getFieldsToShow')` in the function, then it looks like this:
![Bildschirmfoto 2021-08-25 um 09 53 44](https://user-images.githubusercontent.com/463851/130750577-6edca115-adca-4ee8-9e5d-541b10d5a530.png)
so this should be refactored to be calculate the fields to show in `doc_table_wrapper.tsx`. It's working fine using new table aka data grid btw 👍